src/Startup.cs

```diff
@@ -31,6 +32,7 @@ public void ConfigureServices(IServiceCollection services)
                     options.SlidingExpiration = true;
 
                 });
+            services.AddScoped<EmailSender>();
```

Comment:
This is some kind of DI as I understand, but for example in `Register.cshtml.cs` instance of object is created. So shortly how does it work and what does it give? (source to read about it myself is welcome as well).

Reply:
Yeah, this is DI. I'll try to explain it in a few words, but I'm not sure I will be able because it's a bit complex. Basically instead of using constructors everywhere in the model, on it being created the needed object are already passed to the model. We can also add the different lifetime of the objects, and so on. As I understand this is used to make the classes less tightly coupled, as well as more efficient usage of Database contexts for example

Sources I've relied on mostly:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0
https://www.youtube.com/watch?v=HXKGKvoQ0Xs&t=329s
https://stackoverflow.com/questions/35367127/asp-net-mvc-guidelines-for-static-classes-for-database-access